crates/metrics/metrics-derive/src/expand.rs

```diff
@@ -60,6 +66,13 @@ pub(crate) fn derive(node: &DeriveInput) -> Result<proc_macro2::TokenStream> {
                 }
 
                 impl #ty {
+                    /// Create new instance of metrics with provided labels.
+                    #vis fn new_with_labels(labels: impl metrics::IntoLabels + Clone) -> Self {
```

Comment:
Would it make sense to have `new_with_labels` instead take a `Cow<[impl metrics::IntoLabels]>` so we don't have to do `labels.clone()`?